coderd/coderd.go

```diff
@@ -1534,16 +1536,27 @@ func New(options *Options) *API {
 	// browsers, so these don't make sense on api routes.
 	cspMW := httpmw.CSPHeaders(
 		api.Experiments,
-		options.Telemetry.Enabled(), func() []string {
+		options.Telemetry.Enabled(), func() []*proxyhealth.ProxyHost {
 			if api.DeploymentValues.Dangerous.AllowAllCors {
-				// In this mode, allow all external requests
-				return []string{"*"}
+				// In this mode, allow all external requests.
+				return []*proxyhealth.ProxyHost{
+					{
+						Host:    "*",
+						AppHost: "*",
+					},
+				}
+			}
+			// Always add the primary, since the app host may be on a sub-domain.
+			proxies := []*proxyhealth.ProxyHost{
+				{
+					Host:    api.AccessURL.Host,
+					AppHost: appurl.ConvertAppHostForCSP(api.AccessURL.String(), api.AppHostname),
+				},
 			}
```

Comment:
I tacked on the primary here, but we could instead add the primary as part of the `WorkspaceProxyHostsFn` function.

Reply:
`api.AppHostname` can be `""`. We should probably handle empty string, and not add it to the csp header.

Reply:
`ConvertAppHostForCSP` uses the first argument if the second is `""`, so we end up adding the base path instead (my thinking here was that means we are using path-based apps, so the root domain is the app host).